tests/model_serving/model_server/keda/test_isvc_keda_scaling.py

```diff
@@ -0,0 +1,201 @@
+import pytest
+from simple_logger.logger import get_logger
+from typing import Any, Generator
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.pod import Pod
+from utilities.constants import KServeDeploymentType, Timeout
+from tests.model_serving.model_server.utils import (
+    verify_keda_scaledobject,
+    run_vllm_concurrent_load,
+    run_ovms_concurrent_load,
+)
+from utilities.infra import get_pods_by_isvc_label
+from tests.model_serving.model_runtime.vllm.constant import (
+    COMPLETION_QUERY,
+    CHAT_QUERY,
+    BASE_RAW_DEPLOYMENT_CONFIG,
+)
+from tests.model_serving.model_runtime.vllm.basic_model_deployment.test_granite_7b_starter import (
+    SERVING_ARGUMENT,
+    MODEL_PATH,
+)
+from tests.model_serving.model_server.serverless.utils import inference_service_pods_sampler
+from utilities.constants import ModelFormat, ModelVersion, RunTimeConfigs
+from utilities.monitoring import validate_metrics_field
+
+LOGGER = get_logger(name=__name__)
+
+
+BASE_RAW_DEPLOYMENT_CONFIG["runtime_argument"] = SERVING_ARGUMENT
+
+INITIAL_POD_COUNT = 1
+FINAL_POD_COUNT = 5
+
+VLLM_MODEL_NAME = "granite-vllm-keda"
+VLLM_METRICS_QUERY_GPU = f'sum_over_time(vllm:gpu_cache_usage_perc \
+    {{namespace="{VLLM_MODEL_NAME}",pod=~"{VLLM_MODEL_NAME}.*"}}[5m])'
+VLLM_METRICS_THRESHOLD_GPU = 0.02
+VLLM_METRICS_QUERY_REQUESTS = f'vllm:num_requests_running{{namespace="{VLLM_MODEL_NAME}",pod=~"{VLLM_MODEL_NAME}.*"}}'
+VLLM_METRICS_THRESHOLD_REQUESTS = 4
+
+OVMS_MODEL_NAMESPACE = "ovms-keda"
+OVMS_MODEL_NAME = "onnx-raw"
+OVMS_METRICS_QUERY = (
+    f"sum by (name) (rate(ovms_inference_time_us_sum{{"
+    f"namespace='{OVMS_MODEL_NAMESPACE}', name='{OVMS_MODEL_NAME}'"
+    f"}}[5m])) / "
+    f"sum by (name) (rate(ovms_inference_time_us_count{{"
+    f"namespace='{OVMS_MODEL_NAMESPACE}', name='{OVMS_MODEL_NAME}'"
+    f"}}[5m]))"
+)
+OVMS_METRICS_THRESHOLD = 200
+
+pytestmark = [pytest.mark.keda, pytest.mark.usefixtures("skip_if_no_supported_gpu_type", "valid_aws_config")]
+
+
+@pytest.fixture
+def verify_initial_pod_count(unprivileged_client: DynamicClient):
```

Comment:
I'm not sure I understand why these funcs are wrapped as fixtures.
To me, these are just utility functions and aren't required to be marked as fixtures.